sherpa/astro/ui/tests/test_astro_ui_plot.py

```diff
@@ -3672,3 +3672,45 @@ def test_set_ylog_bkg(plot, yscale, clean_astro_ui):
     assert len(axes) == 1
     assert axes[0].xaxis.get_scale() == 'linear'
     assert axes[0].yaxis.get_scale() == yscale
+
+
+@requires_plotting
+def test_pha_model_plot_filter_range_manual_1024(clean_astro_ui):
+    """Check if issue #1024 is fixed.
+
+    The problem was that the PHA model plot was resetting the
+    range so that the notice call ended up not changing anything.
+    Unfortunately this does not show up in the "manual" case,
+    because the data is not grouped, so you have to rely on
+    test_pha_model_plot_filter_range_1024.
+    """
+
+    setup_example(None)
+    ui.set_analysis('energy')
+
+    ui.plot_model()
+    ui.notice(0.77, 1.125)
+
+    f = ui.get_filter()
+    assert f == '0.775000000000:1.210000000000'
+
+
+@pytest.mark.xfail
+@requires_fits
+@requires_data
+@requires_plotting
+def test_pha_model_plot_filter_range_1024(make_data_path, clean_astro_ui):
+    """Check if issue #1024 is fixed.
+
+    Unlike test_pha_model_plot_fitler_range_manual_1024
+    this test does show issue #1024.
+    """
+
+    ui.load_pha(make_data_path('3c273.pi'))
+    ui.set_source(ui.powlaw1d.pl)
+
+    ui.plot_model()
+    ui.notice(0.5, 5)
+
+    f = ui.get_filter()
+    assert f == '0.518300011754:4.869099855423'
```

Comment:
Not necessarily for this PR, because strong comparison of numbers are a little bit of a pain. However, in astropy tests like this have bitten us later when someone added tests on e.g. 32 bit systems, run the tests on a big endian cluster or any other case where rounding errors can be a little different from the 64 bit work station linux that we usually use.
So instead, one might use:
```
assert float(f.split(':')) == pytest.approx(0.518300011754)
```
I'm sure we have other places in the code where that happens, so we would have to fix that anyway if it comes up, but we can avoid adding new cases.